### PR TITLE
Bug 1898045: redeploy CSI Controller Deployment when secret changes

### PR DESCRIPTION
**Pull request by bertinatto**
This PR uses changes from https://github.com/openshift/library-go/pull/968 to redeploy the Controller Deployment when the secret changes.

CC @openshift/storage

### CODE REVIEW

**Comment by bertinatto** [comment]
/hold

https://github.com/openshift/library-go/pull/968 needs to be merged first.

**Comment by bertinatto** [comment]
/hold cancel
library-go PR has been merged.

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -196,3 +200,27 @@ func isCustomCABundleUsed(kubeClient kubeclient.Interface) (bool, error) {
 	_, exists := cloudConfigCM.Data[caBundleKey]
 	return exists, nil
 }
+
+func withSecretHashAnnotation(kubeClient kubeclient.Interface, namespace, secret string) func(*appsv1.Deployment) error {
+	return func(deployment *appsv1.Deployment) error {
+		inputHashes, err := resourcehash.MultipleObjectHashStringMapForObjectReferences(
```
Comment (jsafrane) [inline pkg/operator/starter.go:206]: Would it be possible to use a secret informer? `MultipleObjectHashStringMapForObjectReferenceFromLister` with an informer/lister from `kubeInformersForNamespaces`.

Comment (bertinatto) [reply pkg/operator/starter.go]: Good catch. Done.

**Comment by bertinatto** [comment]
I manually tested the last changes and they worked.

**Pull request merged**